cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -564,9 +591,12 @@ func (s *Supervisor) onOpampConnectionSettings(_ context.Context, settings *prot
 func (s *Supervisor) waitForOpAMPConnection() error {
 	// wait for the OpAMP client to connect to the server or timeout
 	select {
-	case <-s.connectedToOpAMPServer:
+	// This channel is unbuffered, so this will block until we listen on the channel in `OnConnectFunc`.
+	case s.connectedToOpAMPServer <- struct{}{}:
+		close(s.connectedToOpAMPServer)
 		return nil
-	case <-time.After(10 * time.Second):
+	case <-s.clockImpl.After(10 * time.Second):
```

Comment:
Instead of wrapping `time`, could we make the timeout value configurable via a private field on the Supervisor? I imagine we will want to expose the timeout as a configuration option eventually anyway. It will also set the timeout to be a small value like 5ms for tests.

Reply:
Missed your other comment, I see the problem now. Resolving this to continue discussion there.